src/main/ml-modules/root/lib/interceptors/claim-mapping.sjs

```diff
@@ -0,0 +1,59 @@
+"use strict";
+
+const { getTraceHelpers } = require('/lib/util/trace-helpers.sjs');
+
+declareUpdate();
+
+const { trace, traceObject } = getTraceHelpers('claim-mapping');
+
+const employeesAndRelatives = Array.from(cts.doc('/meta/employeesAndRelatives.json').root.ids).map(id => id.toString());
+const psychDiagnoses = Array.from(cts.doc('/meta/psych-diagnoses.json').root.codes.snomed).map(code => code.toString());
+
+traceObject(employeesAndRelatives);
+traceObject(psychDiagnoses);
+
+const employeePsychReader = xdmp.permission('employee-psych-reader', 'read');
+const employeeReader = xdmp.permission('employee-reader', 'read');
+const psychReader = xdmp.permission('psych-reader', 'read');
+const phiReader = xdmp.permission('phi-reader', 'read');
+
+/** Check to see if any of the SNOMED diagnosis codes on the claim are psychological in nature */
+function isPsychClaim(claim) {
+  return claim.envelope.instance.Claim.diagnosis.some(
+    diagnosis => psychDiagnoses.includes(diagnosis.Diagnosis.code__procedureCodeableConcept__SNOMED.toString())
+  );
+}
+
+/** Check to see if the patient referenced by this claim is a current/former HHS employee or a relative thereof */
+function isClaimForEmployeeOrRelative(claim) {
+  return employeesAndRelatives.includes(claim.envelope.instance.Claim.patient.Reference.reference.toString().replace('Patient/', ''));
+}
+
+var contentArray;
+var options;
+
+traceObject(contentArray);
+traceObject(options);
+
+contentArray.forEach(content => {
+  const claim = content.value.root || content.value;
```

Comment:
Can we add a comment on this? when is the root missing and what does that mean?

Reply:
Yeah, I can add a comment to explain what it does. I learned a few weeks ago in the javascript-nerds channel in Slack that a Document object has two ways to access JSON content, `.toObject` and `.root` - the latter is faster because it doesn't invoke a function, and can rival XQuery for pure speed of access.